contracts/marketplace/direct-listings/DirectListingsLogic.sol

```diff
@@ -469,10 +469,21 @@ contract DirectListingsLogic is IDirectListings, ReentrancyGuardLogic, ERC2771Co
                 IERC1155(_assetContract).balanceOf(_tokenOwner, _tokenId) >= _quantity &&
                 IERC1155(_assetContract).isApprovedForAll(_tokenOwner, market);
         } else if (_tokenType == TokenType.ERC721) {
+            address owner;
+            address operator;
+
+            // failsafe for reverts in case of non-existent tokens
+            try IERC721(_assetContract).ownerOf(_tokenId) returns (address _owner) {
```

Comment:
wouldn't this just return the zero address? it doesn't throw in solidity right?

Reply:
Yeah it won't throw. The operator and owner variables are zero address by default. The check below will evaluate to false in case these remain zero due to external calls failing.

Reply:
then why do we need the try catch?

Reply:
The `ownerOf` function does throw an error in some ERC-721 contracts out there. For example, this is the `ownerOf` code in the OpenZeppelin ERC-721 contract:

```solidity
function ownerOf(uint256 tokenId) public view virtual override returns (address) {
        address owner = _owners[tokenId];
        require(owner != address(0), "ERC721: invalid token ID");
        return owner;
}
```